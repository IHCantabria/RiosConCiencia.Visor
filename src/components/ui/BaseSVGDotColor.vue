<template>
  <div class="svg-dot-container">
    <svg class="svg-dot-color" :style="cssVars">
      <rect
        x="0"
        y="0"
        :rx="borderRadius"
        :ry="borderRadius"
        :width="sizePx"
        :height="sizePx"
        style="stroke-opacity: 1; stroke-linejoin: round; stroke-linecap: round; stroke-width: 3; fill-opacity: 0.2; fill-rule: evenodd;"
      />
    </svg>
  </div>
</template>
<script>
export default {
  props: {
    size: {
      type: Number,
      required: false,
      default: () => {
        return 29;
      }
    },
    color: {
      type: String,
      required: true,
      default: () => {
        return "#FFFFFF";
      }
    },
    borderRadius: {
      type: Number,
      required: false,
      default: () => {
        return 10;
      }
    }
  },
  computed: {
    sizePx() {
      return `${this.size}px`;
    },
    cssVars() {
      return {
        "--size": `${this.sizePx}`,
        "--backgroundColor": `${this.color}`
      };
    }
  }
};
</script>
<style lang="scss" scoped>
.svg-dot-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.svg-dot-color {
  width: var(--size);
  height: var(--size);

  rect {
    stroke: var(--backgroundColor);
    fill: var(--backgroundColor);
  }
}
</style>
