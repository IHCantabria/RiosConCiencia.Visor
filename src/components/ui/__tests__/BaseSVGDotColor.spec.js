import { shallowMount } from "@vue/test-utils";
import BaseSVGDotColor from "../BaseSVGDotColor.vue";
import merge from "lodash.merge";

function createWrapper(overrides) {
  const defaultMountingOptions = {
    propsData: { color: "#456789" }
  };
  return shallowMount(
    BaseSVGDotColor,
    merge(defaultMountingOptions, overrides)
  );
}

describe("baseSVGDotColor.vue", () => {
  //SNAPSHOT TESTS
  it("renders correctly the svg with the default values", () => {
    expect.assertions(1);
    const wrapper = createWrapper({ propsData: {} });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly", () => {
    expect.assertions(1);
    const wrapper = createWrapper({
      propsData: {
        size: 15,
        borderRadius: 20
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
