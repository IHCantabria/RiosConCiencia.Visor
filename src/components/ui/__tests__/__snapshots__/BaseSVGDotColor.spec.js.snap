// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseSVGDotColor.vue renders correctly 1`] = `
<div
  class="svg-dot-container"
>
  <svg
    class="svg-dot-color"
    style="--size: 15px; --backgroundColor: #456789;"
  >
    <rect
      height="15px"
      rx="20"
      ry="20"
      style="stroke-opacity: 1; stroke-linejoin: round; stroke-linecap: round; stroke-width: 3; fill-opacity: 0.2; fill-rule: evenodd;"
      width="15px"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`baseSVGDotColor.vue renders correctly the svg with the default values 1`] = `
<div
  class="svg-dot-container"
>
  <svg
    class="svg-dot-color"
    style="--size: 29px; --backgroundColor: #456789;"
  >
    <rect
      height="29px"
      rx="10"
      ry="10"
      style="stroke-opacity: 1; stroke-linejoin: round; stroke-linecap: round; stroke-width: 3; fill-opacity: 0.2; fill-rule: evenodd;"
      width="29px"
      x="0"
      y="0"
    />
  </svg>
</div>
`;
